lms/djangoapps/instructor/utils.py

```diff
@@ -40,6 +47,122 @@ def get_module_for_student(student, usage_key, request=None):
     field_data_cache = FieldDataCache([descriptor], usage_key.course_key, student)
     return get_module(student, request, usage_key, field_data_cache)
 
+
+def collect_course_forums_data(course_id):
+    """
+    Given a SlashSeparatedCourseKey course_id, return headers and information
+    related to course forums usage
+    """
+    try:
+        client = MongoClient(get_mongo_connection_string())
+        mongodb = client[FORUMS_MONGO_PARAMS['database']]
+        new_threads_query = generate_course_forums_query(course_id, "CommentThread")
+        new_responses_query = generate_course_forums_query(course_id, "Comment", False)
+        new_comments_query = generate_course_forums_query(course_id, "Comment", True)
+
+        new_threads = mongodb.contents.aggregate(new_threads_query)['result']
+        new_responses = mongodb.contents.aggregate(new_responses_query)['result']
+        new_comments = mongodb.contents.aggregate(new_comments_query)['result']
+    except PyMongoError:
+        raise
+
+    for entry in new_responses:
+        entry['_id'][u'type'] = "Response"
```

Comment:
Why is type here unicode, but not in line 75? Also it's the only unicode string?